product_pricelist_supplierinfo/models/product_supplierinfo.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2020 Akretion - Mourad EL HADJ MIMOUNE
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ProductSupplierinfo(models.Model):
+    _inherit = "product.supplierinfo"
+
+    sale_margin = fields.Float(
+        default=0,
+        digits=(16, 2),
+        help="Margin to apply on price to obtain sale price",
+    )
+
+    def _get_supplierinfo_pricelist_price(self):
+        self.ensure_one()
+        sale_price = self.price
+        if self.sale_margin:
+            sale_price = (self.price + (self.price * (self.sale_margin / 100))) or 0.0
```

Comment:
Since v17, there is a discount field on supplierinfo and a `price_discounted`
If you want to sale a product depending on the price you buy it, this is the price_discounted that should be used, right ?
Or at least, if not by default, it should be an option in the module ?
What do you think ? @pedrobaeza @JordiBForgeFlow @marielejeune

Reply:
Yeah, I think it should be included in this module.